### ISSUE TITLE
Please support JSON / Object('json') type - at least [bulk] inserts

### ISSUE THREAD

**Issue by v-instrumentix** [ClickHouse/clickhouse-cpp#308]
Dear all,
my app (c++) needs to populate JSON column.
Is there a plan to add support for it?

I made attempt to add JSON following implementation for String type (https://github.com/v-instrumentix/clickhouse-cpp.git - sorry if you find it nasty), but ended with server error

> DB::Exception: Syntax error (data type): failed at position 1 ('{'): {"a":false, "b":true, "c":0, "d":1, "e":"Hello", "d":"", "z":null}. Expected one of: data type, nested table, identifier.

..that looks as I miss something about the protocol / encoding, and this is where I stuck.
Any hints would be welcome.

Thanks!

**Comment by Enmk** [comment]
Hi @v-instrumentix ! Well, there were no plans to support Json type, but any contribution is more than welcome. Please do not forget to update tests as well (I haven't found any in your repo)

**Comment by v-instrumentix** [comment]
Dear @Enmk , thanks for returning back to me.
I would be happy to contribute, but I miss knowledge of client-server protocol and encoding, so I failed with adding support for JSON.
If you could point me to appropriate docs, I'll give it another try.
Cheers!

**Comment by Enmk** [comment]
There are not much docs on native client-server protocol, you'd have to read source of ClickHouse in order to understand how the data is transferred:
- https://github.com/ClickHouse/ClickHouse/blob/master/programs/client/Client.cpp - how packets are sent from client to server
- https://github.com/ClickHouse/ClickHouse/blob/master/src/Server/TCPHandler.cpp - how packets are processed on server side
- https://github.com/ClickHouse/ClickHouse/blob/master/src/Formats/NativeWriter.cpp - low-level protocol serialization/deterialization of ints, strings, etc.
- https://github.com/ClickHouse/ClickHouse/blob/master/src/DataTypes/Serializations/SerializationObject.cpp - higher-level serialization of `Object('json')` to/from native format.

**Comment by v-val** [comment]
@Enmk right, I tried to follow the code and made some progress this way but then stuck with the error above; hoped there're some docs that might help.

FYI, I was able to sort the functional side of problem via plain text bulk insert statements:
```
INSERT INTO `table` FORMAT TabSeparated
...
```

Thanks a lot for the hints anyway, I'll use them if / when I have a chance to proceed with this.

**Comment by slabko** [comment]
JSON is no supported in [version 2.6.2](https://github.com/ClickHouse/clickhouse-cpp/releases/tag/v2.6.2) using `output_format_native_write_json_as_string`. Please see more in the PR description [Experimental support for the JSON data type](https://github.com/ClickHouse/clickhouse-cpp/pull/490)
